### PR TITLE
Update implementors of Iterable, List, and Set to declare new methods.

### PR DESCRIPTION
**Pull request by srawlins**
These new methods will need to be found on implementors in Dart 2.0:
https://gist.github.com/lrhn/b99fe5f73a10b3b1a91579853056d08b

I left everything as unimplemented. I tried to put methods in alphabetical
order, but BuiltList and BuiltSet were in an order that I could not grok.

Feel free to add edits implementing or sorting methods, or comment and I can do some of it. Anything more-than-trivial should probably not be done by me. This just lets a Dart 2.0 user use this package before the package properly implements the new methods, which may or may not take time.

---
This change is [Reviewable](https://reviewable.io/reviews/google/built_collection.dart/119)

### CODE REVIEW

File: lib/src/internal/copy_on_write_list.dart [hunk 1]
```diff
@@ -19,12 +19,26 @@ class CopyOnWriteList<E> implements List<E> {
   @override
   E operator [](int index) => _list[index];
 
+  @override
+  // TODO: Dart 2.0 requires this method to be implemented.
+  // ignore: override_on_non_overriding_method
+  List<E> operator +(List<E> other) {
+    throw new UnimplementedError("+");
```
Comment (davidmorgan) [inline lib/src/internal/copy_on_write_list.dart]: Please use single quotes for strings; everywhere else, too :/

Comment (srawlins) [reply lib/src/internal/copy_on_write_list.dart]: Done!

**Comment by srawlins** [comment]
Thanks much David!

**Comment by davidmorgan** [comment]
Looks good, thanks.

**Pull request merged**